hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -326,6 +326,8 @@ protected void completeCompaction(HoodieCommitMetadata metadata,
             + config.getBasePath() + " at time " + compactionCommitTime, e);
       }
     }
+    autoCleanOnCommit();
+    autoArchiveOnCommit(table, true);
```

Comment:
One table service should not trigger other table services.  Table services should only be triggered by one regular writer.